bokehjs/src/coffee/common/plot.coffee

```diff
@@ -284,12 +307,24 @@ class PlotView extends ContinuumView
     for renderer, i in @mget("renderers")
       indices[renderer.id] = i
     sortKey = (renderer) -> indices[renderer.model.id]
-
+    
+    gl_collections = []
     for level in levels
       renderers = _.sortBy(_.values(@levels[level]), sortKey)
       for renderer in renderers
         renderer.render()
-
+        # If that glyph renders in gl, store its collection.
+        # Note that multiple glyphs can share a collection.
+        if renderer.glyph?
+          glglyph = renderer.glyph.glglyph
+          if glglyph?          
+            if glglyph.coll not in gl_collections              
+              gl_collections.push(glglyph.coll)
+    
+    # Draw collections 
+    for coll in gl_collections
+      coll.glglyphs[0].draw_collection()
```

Comment:
I am not particularly happy with this bit. 

Quick explanation: to make collections work, every glyph gets its _render function called, which calls a function that prepares that glyph for being drawn in the collection. At the very end, the collection is rendered, effectively drawing all glyphs (of the same type) at once.

To have this extra render pass for the collections, I need to modify the rendering mechanics. Would this be the best place? Any suggestions?